src/services/toolbarSettingsService.ts

```diff
@@ -16,19 +16,106 @@
 ///
 
 ///<reference path="../tsd.d.ts"/>
+
+interface IToolbarSettings {
+  items: any[];
+  dataViews: any[];
+}
+
+interface IRequestData {
+  lastAction: string;
+  id?: number;
+  display?: string;
+  gtl_type?: string;
+}
+
 export default class ToolbarSettingsService {
+  private countSelected: number = 0;
+  public items: any[];
+  public dataViews: any[];
+
   /*@ngInject*/
   constructor(private $http: any, private MiQEndpointsService: any) {}
 
-  public getSettings(isList = false) {
+  /**
+   *
+   * @param isClicked
+   */
+  public checkboxClicked(isClicked) {
+    isClicked ? this.countSelected++ : this.countSelected--;
+    _.chain(this.items)
+      .flatten()
+      .each((item: any) => {
```

Comment:
Since you are using a `each/if` pattern you could probably just replace with `filter`